firmware/src/main.cpp

```diff
@@ -1,67 +1,115 @@
+#include <pico/stdlib.h> // for uart printing
 #include <cstring>
+#include <stdio.h>
+#include <config.h>
+#include <pico/multicore.h>
 #include <harp_c_app.h>
 #include <harp_synchronizer.h>
 #include <core_registers.h>
 #include <reg_types.h>
-#include <bme688.h>
-#include <config.h>
-#ifdef DEBUG
-    #include <pico/stdlib.h> // for uart printing
-    #include <cstdio> // for printf
-#endif
+#include "core1_sensor.h"
 
 // Create device name array.
-const uint16_t who_am_i = 1234;
+const uint16_t who_am_i = 1405;
 const uint8_t hw_version_major = 1;
 const uint8_t hw_version_minor = 0;
 const uint8_t assembly_version = 2;
-const uint8_t harp_version_major = 2;
+const uint8_t harp_version_major = 0;
 const uint8_t harp_version_minor = 0;
-const uint8_t fw_version_major = 3;
-const uint8_t fw_version_minor = 0;
+const uint8_t fw_version_major = 0;
+const uint8_t fw_version_minor = 1;
 const uint16_t serial_number = ENV_SENSOR_DEVICE_ID;
```

Comment:
Should the serial number just be 0 if the who_am_i is ENV_SENSOR_DEVICE_ID? That's what other projects seem to use.

Reply:
We're leaving this as 0 for now since Harp is long-term planning to phase it out and replace it with a UUID register. A unique identifier is currently [auto-detected from the HarpCore](https://github.com/AllenNeuralDynamics/harp.core.rp2040/blob/main/firmware/src/harp_core.cpp#L39) using a unique value from the external flash chip, but hasn't been fully merged into the protocol [PR](https://github.com/harp-tech/protocol/pull/39).